src/components/post-block/react/PostBlock.tsx

```diff
@@ -0,0 +1,208 @@
+    thumbnailAspectRatio?: ThumbnailAspectRatio;
+    /* Post title */
+    title: string;
+    /* Theme. */
+    theme?: Theme;
+    /* Callback for the click event. */
+    onClick?(): void;
+}
+type PostBlockProps = IPostBlockProps;
+
+/////////////////////////////
+
+/**
+ * Define the types of the default props.
+ */
+interface IDefaultPropsType extends Partial<PostBlockProps> {}
+
+/////////////////////////////
+//                         //
+//    Public attributes    //
+//                         //
+/////////////////////////////
+
+/**
+ * The display name of the component.
+ */
+const COMPONENT_NAME = `${COMPONENT_PREFIX}Progress`;
+
+/**
+ * The default class name and classes prefix for this component.
+ */
+const CLASSNAME: string = getRootClassName(COMPONENT_NAME);
+
+/**
+ * The default value of props.
+ */
+const DEFAULT_PROPS: IDefaultPropsType = {
+    orientation: Orientation.horizontal,
+    theme: Theme.light,
+    thumbnailAspectRatio: ThumbnailAspectRatio.horizontal,
+};
+/////////////////////////////
+
+/**
+ * Simple Progress component that can be displayed as a linear or circular element
+ *
+ * @return The component.
+ */
+const PostBlock: React.FC<PostBlockProps> = ({
+    actions,
+    attachments,
+    author,
+    meta,
+    onClick,
+    orientation = DEFAULT_PROPS.orientation,
+    tags,
+    text,
+    thumbnail,
+    thumbnailAspectRatio = DEFAULT_PROPS.thumbnailAspectRatio,
+    title,
+    theme = DEFAULT_PROPS.theme,
+}: PostBlockProps): ReactElement => {
+    const [hasActions, setHasActions] = useState(false);
+    const [hasAttachments, setHasAttachments] = useState(false);
+    const [hasAuthor, setHasAuthor] = useState(false);
+    const [hasTags, setHasTags] = useState(false);
+
+    const el = useRef<HTMLDivElement>(null);
+
+    const deleteElementClasses = (classRegex: RegExp): void => {
+        if (!el || !el.current) {
+            return;
+        }
+
+        const classesToDelete: string[] = el.current.classList.value
+            .split(' ')
+            .filter((value: string) => value.match(classRegex));
+
+        for (const classItem of classesToDelete) {
+            el.current.classList.remove(classItem);
+        }
+    };
+
+    useEffect((): void => {
+        if (actions) {
+            setHasActions(true);
+        }
+
+        if (author) {
+            setHasAuthor(true);
+        }
+
+        if (attachments) {
+            setHasAttachments(true);
+        }
+
+        if (tags) {
+            setHasTags(true);
+        }
+    }, [actions, author, tags]);
+
+    useEffect((): void => {
+        if (!el || !el.current) {
+            return;
+        }
+
+        if (!theme) {
+            el.current.classList.add(`${CSS_PREFIX}-post-block--theme-${DEFAULT_PROPS.theme}`);
+        }
+
+        deleteElementClasses(/(?:\S|-)*post-block--orientation-\S+/g);
+
+        el.current.classList.add(`${CSS_PREFIX}-post-block--orientation-${orientation}`);
+    }, [el, theme]);
```

Comment:
You don't need `useEffect` or the element reference here to set the class names on the component.
Please take on other components like the UserBlock.
Here is an example code:
```
<div
            className={classNames(
                className,
                handleBasicClasses({ prefix: CLASSNAME, orientation, theme }),
            )}
>
```
This will generate the HTML => `<div class="lumx-post-block lumx-post-block--theme-light lumx-post-block--orientation-vertical">` for theme=light and orientation=vertical